### ISSUE TITLE
pinpoint does not support spring cloud gateway ? 

### ISSUE THREAD

**Issue by dean-hu** [naver/pinpoint#5267]
Hello,
I found in use that pinpoint could not collect data from the SPIRNG Cloud Gateway project (Spring Cloud Gateway is a gateway service through which all requests are routed to specific microservices with an official address of Https://spring.io/projects/spring-cloud-gateway). But for the spring boot project, it can be collected.

**Comment by RoySRose** [comment]
@dean-hu 

You got it right~
Would be great if you can contribute a plug-in on this :)

**Comment by dean-hu** [comment]
> @dean-hu
> 
> You got it right~
> Would be great if you can contribute a plug-in on this :)

Yeah I understand very much your reply. I really want to provide such a plugin to support the spring cloud gateway, but I still don't know how to do it. Do you intend to implement this function in the near future?

**Comment by RoySRose** [comment]
@dean-hu 

Currently, we don't have any plans for it.
Most of the team members are planned to focus on something else for this year.

Developing a plug-in isn't so complicated. 
You can start from [here](https://naver.github.io/pinpoint/plugindevguide.html)
It may look like there is a lot to understand before you can start. 
But eventually, it's just to find where to instrument.

**Comment by 247687009** [comment]
I also need this plugin~

**Comment by oatiz** [comment]
+1, also need this plugin~

**Comment by crab890715** [comment]
+1, also need this plugin~

**Comment by jasoncool520** [comment]
+1, also need this plugin~

**Comment by zhougui** [comment]
I also need this plugin~

**Comment by monkeyAlwaysWin** [comment]
I also need this plugin~

**Comment by armstrongwang5178** [comment]
I also need this plugin~  When can I monitor the gateway？？

**Comment by guoyuzhuang** [comment]
+1, also need this plugin~

**Comment by guoyuzhuang** [comment]
> Hello,
> I found in use that pinpoint could not collect data from the SPIRNG Cloud Gateway project (Spring Cloud Gateway is a gateway service through which all requests are routed to specific microservices with an official address of [Https://spring.io/projects/spring-cloud-gateway](Https://spring.io/projects/spring-cloud-gateway)). But for the spring boot project, it can be collected.

Have you solved the problem? How was it solved? Can you share it? My email is guoyuzhuang@126.com ，Looking forward to your reply

**Comment by SoonhyukYoon** [comment]
+1, also need this plugin~

**Comment by InkTeam** [comment]
+1, also need this plugin~

**Comment by cailumin** [comment]
+1, also need this plugin~

**Comment by maxuewen** [comment]
+1, also need this plugin~

**Comment by mofazhiku** [comment]
+10086, also still need this plugin~

**Comment by emeroad** [comment]
Pinpoint already supports spring cloud gateway.
Since spring cloud gateway uses webflux,
Webflux support means support of spring cloud gateway.

**Comment by seoah-lab** [comment]
I'm used spring-cloud-starter-gateway:3.1.6 but pinpoint not support transaction.
What settings should I add? @emeroad

**Comment by seoah-lab** [comment]
Self-answer
I used undertow. maybe some setting for undertow.
netty is support! let's check webserver :see_no_evil:

**Comment by emeroad** [comment]
See webflux configuration below.
https://github.com/pinpoint-apm/pinpoint/tree/master/plugins/spring-webflux

**Comment by Robot1024** [comment]
@emeroad 
the version of  pinpoint   master 2.5.1
                            
  user ---> appA  ----> spring cloud gateway(filter) ---->appB

use pinpoint to  monitor  spring cloud gateway、appB

the  display  spring cloud gateway have  data , but  appB  no  data .

if i request by browser  to  appB , then  appB  has the data.

The prinpoint.config
------------------------------------------------------
###########################################################
# Spring WebFlux
###########################################################
profiler.spring.webflux.enable=true

# Client
# If you are not using Reactor-Netty, set it to true.
profiler.spring.webflux.client.enable=true
# Record Parameter.
profiler.spring.webflux.client.param=true
# Record cookies.
profiler.spring.webflux.client.cookie=true
# When cookies should be dumped. It could be ALWAYS or EXCEPTION.
profiler.spring.webflux.client.cookie.dumptype=ALWAYS
# 1 out of n cookies will be sampled where n is the rate. (1: 100%)
profiler.spring.webflux.client.cookie.sampling.rate=1
# Cookie dump size.
profiler.spring.webflux.client.cookie.dumpsize=1024

--------------------------------------------------------

###########################################################
# Reactor Netty
###########################################################
profiler.reactor-netty.enable=true

# Classes for detecting application server type. Comma separated list of fully qualified class names. Wildcard not supported.
profiler.reactor-netty.server.bootstrap.main=
# use an asynchronous endpoint.
profiler.reactor-netty.server.end-point.async.enable=true
# trace param in request  ,default value is true
profiler.reactor-netty.server.tracerequestparam=true
# URLs to exclude from tracing.
# Support ant style pattern. e.g. /aa/*.html, /??/exclude.html
profiler.reactor-netty.server.excludeurl=
# HTTP Request methods to exclude from tracing
#profiler.reactor-netty.server.excludemethod=

# original IP address header
# https://en.wikipedia.org/wiki/X-Forwarded-For
#profiler.reactor-netty.server.realipheader=X-Forwarded-For
# nginx real ip header
#profiler.reactor-netty.server.realipheader=X-Real-IP
# optional parameter, If the header value is ${profiler.reactor-netty.realipemptyvalue}, Ignore header value.
#profiler.reactor-netty.server.realipemptyvalue=unknown

--------------------------------------------------
###########################################################
# Netty
###########################################################
# recommend netty plugin disable, when using VERTX.
profiler.netty=true
profiler.netty.http=true

profiler.netty.http.param=true
# use channel.close()
profiler.netty.channel.close=true
